diff -Nur device-mapper.1.02.08/lib/ioctl/libdm-iface.c device-mapper.1.02.08.force-local-headers/lib/ioctl/libdm-iface.c
--- device-mapper.1.02.08/lib/ioctl/libdm-iface.c	2006-08-02 01:59:16.000000000 +0200
+++ device-mapper.1.02.08.force-local-headers/lib/ioctl/libdm-iface.c	2006-08-02 01:59:17.000000000 +0200
@@ -36,7 +36,7 @@
 #  define MKDEV(x,y) makedev((x),(y))
 #endif
 
-#include <linux/dm-ioctl.h>
+#include "../../kernel/ioctl/dm-ioctl.h"
 
 /*
  * Ensure build compatibility.  
