--- dm/configure.in.org	2006-04-22 20:35:30.751542250 +0200
+++ dm/configure.in	2006-04-22 20:35:40.452148500 +0200
@@ -234,7 +234,7 @@
 
 ################################################################################
 dnl -- Check for getopt
-AC_CHECK_HEADERS(getopt.h, LVM_DEFS="$LVM_DEFS -DHAVE_GETOPTLONG")
+AC_CHECK_HEADERS(getopt.h)
 
 ################################################################################
 dnl -- Internationalisation stuff
