diff -ur device-mapper.0.96.08/dmsetup/Makefile.in device-mapper.0.96.08.bzz/dmsetup/Makefile.in
--- device-mapper.0.96.08/dmsetup/Makefile.in	Wed Jan  2 14:40:48 2002
+++ device-mapper.0.96.08.bzz/dmsetup/Makefile.in	Tue Apr 15 17:38:28 2003
@@ -17,7 +17,7 @@
 	$(CC) -o dmsetup dmsetup.o $(LD_FLAGS) -ldevmapper
 
 install: dmsetup
-	$(INSTALL) -D -o $(OWNER) -g $(GROUP) -m 555 $(STRIP) $< $(sbindir)/$<
+	$(INSTALL) -D -m 555 $(STRIP) $< $(DESTDIR)$(sbindir)/$<
 
 .PHONY: install 
 
diff -ur device-mapper.0.96.08/lib/Makefile.in device-mapper.0.96.08.bzz/lib/Makefile.in
--- device-mapper.0.96.08/lib/Makefile.in	Tue Jan 21 22:22:55 2003
+++ device-mapper.0.96.08.bzz/lib/Makefile.in	Tue Apr 15 17:37:46 2003
@@ -20,17 +20,17 @@
 
 install: install_@interface@
-	$(LN_S) -f libdevmapper.so.$(LIB_VERSION) $(libdir)/libdevmapper.so
-	$(INSTALL) -D -o $(OWNER) -g $(GROUP) -m 444 libdevmapper.h \
-		$(includedir)/libdevmapper.h
+	$(LN_S) -f /lib/libdevmapper.so.$(LIB_VERSION) $(DESTDIR)$(libdir)/libdevmapper.so
+	$(INSTALL) -D -m 444 libdevmapper.h \
+		$(DESTDIR)$(includedir)/libdevmapper.h
 
 .PHONY: install install_@interface@
 
 install_fs: fs/libdevmapper.so
-	$(INSTALL) -D -o $(OWNER) -g $(GROUP) -m 555 $(STRIP) $< \
-		$(libdir)/libdevmapper.so.$(LIB_VERSION)
+	$(INSTALL) -D -m 555 $(STRIP) $< \
+		$(DESTDIR)$(libdir)/libdevmapper.so.$(LIB_VERSION)
 
 install_ioctl: ioctl/libdevmapper.so
-	$(INSTALL) -D -o $(OWNER) -g $(GROUP) -m 555 $(STRIP) $< \
-		$(libdir)/libdevmapper.so.$(LIB_VERSION)
+	$(INSTALL) -D -m 555 $(STRIP) $< \
+		$(DESTDIR)$(libdir)/libdevmapper.so.$(LIB_VERSION)
 
 distclean_lib: 
diff -ur device-mapper.0.96.08/man/Makefile.in device-mapper.0.96.08.bzz/man/Makefile.in
--- device-mapper.0.96.08/man/Makefile.in	Wed Jan  2 14:40:49 2002
+++ device-mapper.0.96.08.bzz/man/Makefile.in	Tue Apr 15 17:37:55 2003
@@ -30,6 +30,6 @@
 	@for f in $(MANUALS); \
 	do \
 	   $(RM) $(MAN8DIR)/$$f; \
-	   @INSTALL@ -D -o $(OWNER) -g $(GROUP) -m 444 $$f $(MAN8DIR)/$$f; \
+	   @INSTALL@ -D -m 444 $$f $(DESTDIR)$(MAN8DIR)/$$f; \
 	done
 
